From 950311506becfb7b21320e8ec0ef67c9e8082871 Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Wed, 18 Jun 2014 09:58:14 -0400
Subject: [net] sctp: fix a missed .data initialization

Message-id: <553af5297ff0326dde26b4b88916a27eb2697774.1403084956.git.dborkman@redhat.com>
Patchwork-id: 84205
O-Subject: [RHEL7 PATCH net 3/5] sctp: fix a missed .data initialization
Bugzilla: 1110290
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110290
Upstream Status: 22a1f5140ed69baa5906ce9b2b9143478f5a4da0
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7587477
Tested: verified on beaker

Upstream Commit:
commit 22a1f5140ed69baa5906ce9b2b9143478f5a4da0
Author: wangweidong <wangweidong1@huawei.com>
Date:   Wed Feb 12 09:44:43 2014 +0800

    sctp: fix a missed .data initialization

    As commit 3c68198e75111a90("sctp: Make hmac algorithm selection for
     cookie generation dynamic"), we miss the .data initialization.
    If we don't use the net_namespace, the problem that parts of the
    sysctl configuration won't be isolation and won't occur.

    In sctp_sysctl_net_register(), we register the sysctl for each
    net, in the for(), we use the 'table[i].data' as check condition, so
    when the 'i' is the index of sctp_hmac_alg, the data is NULL, then
    break. So add the .data initialization.

    Acked-by: Neil Horman <nhorman@tuxdriver.com>
    Signed-off-by: Wang Weidong <wangweidong1@huawei.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sctp/sysctl.c b/net/sctp/sysctl.c
index daea130..a63b3c8 100644
--- a/net/sctp/sysctl.c
+++ b/net/sctp/sysctl.c
@@ -158,6 +158,7 @@ static struct ctl_table sctp_net_table[] = {
  },
  {
   .procname = "cookie_hmac_alg",
+  .data  = &init_net.sctp.sctp_hmac_alg,
   .maxlen  = 8,
   .mode  = 0644,
   .proc_handler = proc_sctp_do_hmac_alg,
-- 
1.7.1